test/transcoding/SPV_INTEL_hw_thread_queries/intel_hw_thread_queries.ll

```diff
@@ -0,0 +1,39 @@
+; Test for SPV_INTEL_hw_thread_queries feature
+; SPIR-V friendly LLVM IR to SPIR-V translation and vice versa
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_hw_thread_queries -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
```

Comment:
I guess the new built-ins don't have any form that is not SPIR-V friendly IR, right? I wonder what is the purpose of checking it twice?

Reply:
Originally, there was no such check in the patch. But then I noticed, that during translation to OCL - the built-ins are not mangled, and I actually need to add extra code for reverse translation. So I added extra checks to reflect this.